servers/lib/test/unit/files.resolver.unit.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { FilesResolver } from "../../src/files/files.resolver";
+import {
+  testDirectory,
+  pathToTestDirectory,
+  pathToTestFileContent,
+  testFileContent,
+} from "../testUtil";
+import { IFilesService } from "../../src/files/interfaces/files.service.interface";
+import { FilesServiceFactory } from "../../src/files/services/files-service.factory";
+
+describe("Unit tests for FilesResolver", () => {
+  let filesResolver: FilesResolver;
+  let filesService: IFilesService;
+
+  beforeEach(async () => {
+    const mockFilesService: IFilesService = {
+      listDirectory: jest
+        .fn()
+        .mockImplementation((pathToTestDirectory) => testDirectory),
+      readFile: jest
+        .fn()
+        .mockImplementation((pathToTestFileContent) => testFileContent),
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        FilesResolver,
+        FilesServiceFactory,
+        {
+          provide: FilesServiceFactory,
+          useValue: {
+            create: () => mockFilesService,
+          },
+        },
+      ],
+    }).compile();
+
+    filesResolver = module.get<FilesResolver>(FilesResolver);
+    filesService = module
+      .get<FilesServiceFactory>(FilesServiceFactory)
+      .create();
+  });
+
+  it("should be defined", () => {
+    expect(filesResolver).toBeDefined();
+  });
+
+  describe("listDirectory", () => {
```

Comment:
@prasadtalasila - multiple of these codeclimate issues want me to refactor my test cases, however it is my understand that it is best practice to keep most of the test data and the actual functionality as simple and homogenous, so that it is easy to understand by looking at the test case without having to look into other files. I could refactor it so that i have these "create test" classes, however, i feel like this would only increase the complexity of understanding the purpose to the test cases, and thus not have an actual benefit. Let me know what you think.